### DISCUSSION TITLE
Is it possible to get the trim matrix so that I can apply it to my orinial image?

### DISCUSSION THREAD

**Discussion by Sicos1977** [dlemstra/Magick.NET#1013] [Help] [answered]
Hi,

I have the code below that I use to convert a color image to black and white and then trim away the black borders around the page. The reason why I add the black border is to prevent a white page getting trimmed to almost nothing.

Is it somehow possible to get the information that is trimmed of the page so that I also can do that with the original color image (that is just a photo of a piece of paper)? 

I can do it on the color image for the deskew option because this report back the angle that is has deskewed my image.

**Comment by dlemstra** [comment]
You can use the `image.BoundingBox` property for this. I am now working on a change that this will become non nullable in the future.

**Comment by Sicos1977** [comment]
BoundingBox does not exactly what I want.

Below is a piece of my original image that is converted to black and white and then the blackborders are trimmed, 
![image](https://user-images.githubusercontent.com/6692947/131527567-066601ff-485c-43f5-ae59-8593be8d8a19.png)


After that I get the boundingbox and apply that to the original color image by cropping it but yes the height and weight are correct but the border is not cropped of but instead the right side is cropped.
![image](https://user-images.githubusercontent.com/6692947/131527718-d8bbb201-06e4-4419-8fd1-377154517bb7.png)


I can't show the complete picture because there is personal information on it.

I expected to have exactly the same image as in black and white but now in color.

Somehow I need to figure out the left and top position from the BoundingBox in relation to the original width and height of the image.

**Comment by Sicos1977** [comment, marked as answer]
Got it I need to get the Page.X and Page.Y and overrule that in the BoundingBox, this way I get exactly what I want.

```c#
Image.Trim(new Percentage(0));
_boundingBox = Image.BoundingBox;
_boundingBox.X = Image.Page.X;
_boundingBox.Y = Image.Page.Y;
OriginalImage.Crop(_boundingBox);
OriginalImage.Write(new FileInfo("d:\\test.jpg"));
```
